SwiftUI Views: Label (#1252)
Codecademy/docs: content/swiftui/concepts/views/terms/Label/label.md

---
Title: 'Label'

> Description: 'Allows users to create a standardised label for buttons.'

Reviewer: ```suggestion
Description: 'A label is a standard user interface component that includes an icon and a title.'
```

Subjects:
  - 'Software Development'
  - 'Mobile Development'
  - 'Computer Science'
Tags:
  - 'SwiftUI'
  - 'SwiftUI Views'
  - 'Label'
  - 'Views'
CatalogContent:
  - 'learn-swiftui'
  - 'paths/build-ios-apps-with-swiftui'
---

> A **label** is a standard user interface component that includes an icon and a title. It appears in many different context like collections, lists or  disclosable lists, buttons, to name a few.

Reviewer: ```suggestion
A **`label`** is a standard user interface component that includes an icon and a title. It appears in many different context like collections, lists or buttons, to name a few.
```

## Syntax

```pseudo
Label("Lightning", systemImage: "bolt.fill")

> }

Reviewer 2: ```suggestion
```

```

> The `label` can be a stand-alone component. It has modifiers where it can display just the icon, text, both or automatically adjust based on the space available.

Reviewer 2: ```suggestion
A `Label` can be a stand-alone component. It has modifiers where it can display just the icon, text, both or automatically adjust based on the space available.
```

> The `"Label"` can be a stand-alone component but the get the most of it is always best to give it a purpose. It has a couple of modifiers where it can display just the icon, text, both or automatically adjust based on the space available.

Reviewer: ```suggestion
The `label` can be a stand-alone component. It has modifiers where it can display just the icon, text, both or automatically adjust based on the space available.
```

## Example

> The following example displays a `Text` that represents a number and two `Buttons`. The first one is using a `Label` and the second one is a broken down version of what `Label` actually does:

Reviewer 2: ```suggestion
The following example displays a `Text` view that represents a number and two `Buttons`. The first one is using a `Label` and the second one is a broken down version of what `Label` actually does:
```

> The following example displays a `Text` that represent a number and two `Buttons`. The first one is using a `Label` and the second one is a broken down version of what `Label` actually does:

Reviewer: ```suggestion
The following example displays a `Text` that represents a number and two `Buttons`. The first one is using a `Label` and the second one is a broken down version of what `Label` actually does:
```

```swift
@State private var score = 0

var body: some View {
    VStack(spacing: 20) {
        Text("\(score)")

        Button {
            score += 1
        } label: {
            Label("Add", systemImage: "plus")
        }

        Button {
            score -= 1
        } label: {
            HStack {
                Image(systemName: "minus")
                Text("Remove")
            }
        }
    }
}
```

> Each time the user presses one of the two `Buttons`, the number above will increase or reduce the score by 1.

Reviewer 2: ```suggestion

Each time the user presses one of the two `Button` views, the `score` above will increase or reduce by `1`.
```

> Each time the user selects press one of the two `Button`, the number above the button `"Add"` will increase or reduce the score by 1.

Reviewer: ```suggestion
Each time the user presses one of the two `Buttons`, the number above will increase or reduce the score by 1.
```

This will display the following:

> ![Label](https://raw.githubusercontent.com/Codecademy/docs/main/media/labels.png)

Reviewer 2: Could we turn this into a gif to show the increment/decrement? If you need assistance with creating a gif I can help you out, just let me know!

Author: Done! The gif is in media and line 61 is updated.
